packages/actor-query-operation-wrap-stream/lib/ActorQueryOperationWrapStream.ts

```diff
@@ -0,0 +1,74 @@
+import type { MediatorProcessIterator } from '@comunica/bus-process-iterator';
+import type {
+  IActionQueryOperation,
+  IActorQueryOperationArgs,
+  MediatorQueryOperation,
+} from '@comunica/bus-query-operation';
+import { ActorQueryOperation, KEY_CONTEXT_WRAPPED_QUERY_OPERATION } from '@comunica/bus-query-operation';
+import type { IActorTest } from '@comunica/core';
+import type { IQueryOperationResult } from '@comunica/types';
+import type * as RDF from '@rdfjs/types';
+import type { AsyncIterator } from 'asynciterator';
+
+/**
+ * A comunica Wrap Stream Query Operation Actor.
+ */
+export class ActorQueryOperationWrapStream extends ActorQueryOperation {
+  public readonly mediatorProcessIterator: MediatorProcessIterator;
+  public readonly mediatorQueryOperation: MediatorQueryOperation;
+
+  public constructor(args: IActorQueryOperationWrapStreamArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionQueryOperation): Promise<IActorTest> {
+    if (action.context.get(KEY_CONTEXT_WRAPPED_QUERY_OPERATION)) {
+      throw new Error('Unable to wrap query source multiple times');
+    }
+    // Ensure this is always run if not already wrapped
+    return { httpRequests: Number.NEGATIVE_INFINITY };
+  }
+
+  public async run(action: IActionQueryOperation): Promise<IQueryOperationResult> {
+    // Prevent infinite recursion. In consequent query operation calls this key should be set to false
+    // To allow the operation to wrap ALL query operation runs
+    action.context = ActorQueryOperation.setContextWrapped(action.context, true);
+    const output: IQueryOperationResult = await this.mediatorQueryOperation.mediate(action);
+    switch (output.type) {
+      case 'bindings':
+        output.bindingsStream = <AsyncIterator<RDF.Bindings>>
```

Comment:
I wonder if we also should allow the metadata to be optionally processed?

This could be useful if someone were to add a process iterator actor that does some filtering, and thereby could reduce the cardinality and update it in the metadata.